cadc-util/src/main/java/ca/nrc/cadc/net/HttpTransfer.java

```diff
@@ -429,6 +460,110 @@ public String getContentMD5() {
     public Date getLastModified() {
         return lastModified;
     }
+
+    /**
+     * Disgest checksum from http header.
+     * @return digest or null
+     */
+    public URI getDigest() {
+        return parseDigest(this.digest);
+    }
+
+    /**
+     * Parse the given digest and return a URI of the form: {algorithm}:{checksum in hex}
+     * @param digest value to parse
+     * @return digest URI
+     */
+    public static URI parseDigest(String digest) {
+        if (!StringUtil.hasLength(digest)) {
+            return null;
+        }
+
+        int index = digest.indexOf('=');
+        if (index == -1) {
+            throw new IllegalArgumentException(String.format("Expected digest format <algorithm>=<checksum value> "
+                                                                 + "found %s", digest));
+        }
+        String scheme = digest.substring(0, index).trim();
+        String checksumValue = digest.substring(index + 1).trim();
+
+        if (scheme.length() == 0) {
+            throw new IllegalArgumentException(String.format("Unable to parse algorithm, expected format "
+                                                                 + "<algorithm>=<checksum value> found %s",
+                                                             digest));
+        }
+        if (checksumValue.length() == 0) {
+            throw new IllegalArgumentException(String.format("Unable to parse checksum, expected format "
+                                                                 + "<algorithm>=<checksum value> found %s",
+                                                             digest));
+        }
+
+        Algorithm algorithm;
+        try {
+            algorithm = Algorithm.fromString(scheme);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(String.format("Unsupported algorithm %s, supported algorithm's: "
+                                                                 + "md5, sha-1, sha-256, sha-384, sha-512", scheme));
```

Comment:
Just a minor point: this list of algorithms will need to be maintained to match the enum, so perhaps remove the list or have it come from the enum.

Reply:
Yes, the list of algorithms should not be hard coded in the exception message. Will skip the list, or get from the enum.